app/src/main/java/org/greenstand/android/TreeTracker/signup/SignupViewModel.kt

```diff
@@ -59,16 +68,22 @@ class SignupViewModel(private val users: Users) : ViewModel() {
 
     fun updateEmail(email: String) {
         // TODO validate data and show errors if needed, after click
-        _state.value = _state.value?.copy(credential = Credential.Email().apply { text = email })
+        _state.value = _state.value?.copy(email = email)
+        _credentialState.value = _credentialState.value?.copy(isEmailValid = Validation.isEmailValid(email))
```

Comment:
Updating two states like this will make the view update twice. This is one of the reasons why we should use 1 state per screen.